website/docs/Use-Cases/Auto-Generation.md

```diff
@@ -424,6 +424,47 @@ In the example above, we create an AssistantAgent named "assistant" to serve as
 3. Once the user_proxy receives a response from the assistant, it tries to reply by either soliciting human input or preparing an automatically generated reply. In this specific example, since `human_input_mode` is set to `"NEVER"`, the user_proxy will not solicit human input but prepare an automatically generated reply (auto reply). More specifically, the user_proxy executes the code and uses the result as the auto-reply.
 4. The assistant then generates a further response for the user_proxy. The user_proxy can then decide whether to terminate the conversation. If not, steps 3 and 4 are repeated.
 
+`UserProxyAgent` can be used to handle function calls to external API, corresponding to the newly added [function calling capability of OpenAI's Chat Completions API](https://openai.com/blog/function-calling-and-other-api-updates?ref=upstract.com). It also allows calling to a function from a class instance. The following code snippet shows how to use `UserProxyAgent` to call a function from a class instance:
+```python
+# define a class
+class AddNum:
+    def __init__(self, given_num):
+        self.given_num = given_num
+
+    def add(self, num_to_be_added):
+        self.given_num = num_to_be_added + self.given_num
+        return self.given_num
+
+# pass in the corresponding class instance and function name
+user = UserProxyAgent(
+    name="test",
+    function_map={
+        "add_num": # this name should match the name in the OpenAI config
+        {
+            "class": AddNum(given_num=10), # initialize the class instance
+            "func_name": "add", # the actual function name in the class
+        }
+    },
+)
+
+# oai_config = {
+#     "function": [
+#         {
+#             name: "add_num",
+#             ...
+#         }
+#     ]
+# }
+
+# below is for demonstration only. In practice, the function call is triggered by the assistant agent.
+func_call = {"name": "add_num", "arguments": '{ "num_to_be_added": 5 }'}
+
+user._execute_function(func_call) # this will call AddNumClass.add_num_func(5) and return 15
+user._execute_function(func_call) # this will call AddNumClass.add_num_func(5) and return 20. The same class instance is used, and the state is preserved.
+```
+
+The code above is used to demonstrate the function calling feature of `UserProxyAgent`. In practice, the function call is triggered by the assistant agent. The assistant agent will send a message to the user proxy agent, which contains the function call information. The user proxy agent will then call the function and return the result to the assistant agent. Please find practical examples in the function call notebook below.
```

Comment:
It'll be hard to find that example far away.

Reply:
I am rewind changes to Auto-Generation.md and change it in a separate PR if needed.